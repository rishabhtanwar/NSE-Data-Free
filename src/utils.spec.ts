import { getDataSchema } from './utils'
describe('Utils', () => {
    const data={
        a: 1,
        b: 'b',
        c: [1,2,3],
        d: {a:1,b:[2,3,4],c:{d:1},e:[]},
        e: new Date('2022-09-12'),
        f: [[1,2],[3,4]]
    }
    test('getDataSchema with type strict',()=>{
        expect(getDataSchema(data)).toMatchSnapshot()
        expect(getDataSchema(['hi'])).toMatchSnapshot('array_validation')
    })
    test('getDataSchema for date value having typeStrict:false',()=>{
        expect(getDataSchema(data,false)).toMatchSnapshot()
        expect(getDataSchema(['hi'],false)).toMatchSnapshot('array_validation')
    })
})
