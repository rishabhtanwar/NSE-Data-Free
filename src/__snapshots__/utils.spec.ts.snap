// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Utils getDataSchema for date value having typeStrict:false 1`] = `
Array [
  "a: any",
  "b: any",
  Object {
    "c": "any",
  },
  Object {
    "d": Array [
      "a: any",
      Object {
        "b": "any",
      },
      Object {
        "c": Array [
          "d: any",
        ],
      },
      Object {
        "e": "any",
      },
    ],
  },
  "e: any",
  Object {
    "f": "any",
  },
]
`;

exports[`Utils getDataSchema for date value having typeStrict:false: array_validation 1`] = `"any"`;

exports[`Utils getDataSchema with type strict 1`] = `
Array [
  "a: number",
  "b: string | null",
  Object {
    "c": "number",
  },
  Object {
    "d": Array [
      "a: number",
      Object {
        "b": "number",
      },
      Object {
        "c": Array [
          "d: number",
        ],
      },
      Object {
        "e": Array [],
      },
    ],
  },
  "e: Date",
  Object {
    "f": "number[]",
  },
]
`;

exports[`Utils getDataSchema with type strict: array_validation 1`] = `"string[]"`;
